Editor/Gui/UiSymbolJson.cs

```diff
@@ -321,33 +315,58 @@ public static SymbolUi ReadSymbolUi(JToken mainObject)
             }
 
             var annotationDict = new OrderedDictionary<Guid, Annotation>();
-            var annotationsArray = (JArray)mainObject["Annotations"];
+            var annotationsArray = (JArray)mainObject[JsonKeys.Annotations];
             if (annotationsArray != null)
             {
                 foreach (var annotationEntry in annotationsArray)
                 {
                     var annotation = new Annotation
                                          {
-                                             Id = Guid.Parse(annotationEntry["Id"].Value<string>()),
-                                             Title = annotationEntry["Title"].Value<string>(),
-                                             PosOnCanvas = (Vector2)vector2Converter(annotationEntry["Position"])
+                                             Id = Guid.Parse(annotationEntry[JsonKeys.Id].Value<string>()),
+                                             Title = annotationEntry[JsonKeys.Title].Value<string>(),
+                                             PosOnCanvas = (Vector2)vector2Converter(annotationEntry[JsonKeys.Position])
                                          };
 
-                    if (annotationEntry["Color"] != null)
+                    var colorEntry = annotationEntry[JsonKeys.Color];
+                    if (colorEntry != null)
                     {
-                        annotation.Color = new Color((Vector4)vector4Converter(annotationEntry["Color"]));
+                        annotation.Color = new Color((Vector4)vector4Converter(colorEntry));
                     }
 
-                    annotation.Size = (Vector2)vector2Converter(annotationEntry["Size"]);
+                    annotation.Size = (Vector2)vector2Converter(annotationEntry[JsonKeys.Size]);
                     annotationDict[annotation.Id] = annotation;
                 }
             }
 
-            var newSymbolUi = new SymbolUi(symbol, symbolChildUis, inputDict, outputDict, annotationDict)
+            symbolUi = new SymbolUi(symbol, symbolChildUis, inputDict, outputDict, annotationDict)
                                   {
-                                      Description = mainObject["Description"]?.Value<string>()
+                                      Description = mainObject[JsonKeys.Description]?.Value<string>()
                                   };
-            return newSymbolUi;
+            return true;
+        }
+
+        private static readonly Type Vector2Type = typeof(Vector2);
+        private static readonly Type Vector4Type = typeof(Vector4);
+        private static readonly Type StylesType = typeof(SymbolChildUi.Styles);
+        private static readonly Type ConnectionStylesType = typeof(SymbolChildUi.ConnectionStyles);
+

```

Comment:
Cached important type info instead of unnecessarily doing it at runtime

Reply:
I did measure this, but I can't tell you exactly what the result was. I basically just A-B'd the total starting time and I remember it being more significant than I even expected. Doing a quick "benchmark" online showed an 8x speedup in type comparison operations, though I wouldn't trust that entirely lmfao